zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -236,7 +236,254 @@ paths:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "401":
+          description: "Unauthorized"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "403":
+          description: "Forbidden"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "404":
+          description: "Bad request"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "500":
+          description: "Internal server error"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+    get:
+      tags:
+        - User
+      summary: "Retrieve details of a user"
+      operationId: "findUserById"
+      parameters:
+        - name: "id"
+          in: "path"
+          required: true
+          schema:
+            type: "integer"
+            format: "int64"
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/CamundaUserDto"
+        "400":
+          description: "Bad request"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "401":
+          description: "Unauthorized"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "403":
+          description: "Forbidden"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "404":
+          description: "Bad request"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "500":
+          description: "Internal server error"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+    put:
+      tags:
+        - User
+      summary: "Update a user"
+      operationId: "updateUser"
+      parameters:
+        - name: "id"
+          in: "path"
+          required: true
+          schema:
+            type: "integer"
+            format: "int64"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CamundaUserWithPasswordDto"
+        required: true
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/CamundaUserDto"
```

Comment:
🔧 Can we try to be consistent with the naming of objects? We don't use `...Dto` in the other objects and I would refrain from using it for the identity-related objects as well if possible.

Reply:
Yeah, I see your point. We have similar "conflicts" in other controllers as well. Currently, it looks like we decided to "live" with it and use explicit class names including the package for one of the classes when both are needed in the same class.

Naming-wise, I would usually recommend something like `CamundaUserRequest` and `CamundaUserResponse` fo the REST API, for example. That way, it doesn't have technical `...DTO` name, it clearly states if this goes in or out of the API and clearly separates itself from the internal classes. WDYT?

Reply:
>Naming-wise, I would usually recommend something like CamundaUserRequest and CamundaUserResponse fo the REST API, for example. That way, it doesn't have technical ...DTO name, it clearly states if this goes in or out of the API and clearly separates itself from the internal classes. WDYT?

That's a really nice idea, I like it very much! ... Of course, `...Request` and `...Response` are super clear suffixes 🥳 Neat!

Reply:
...and done :)